src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -645,10 +650,35 @@ private void SetOperationStarted()
         {
             if (!_operationStarted)
             {
-                // Since this is the first operation, we set the proxy and server credentials on the
-                // WinRT filter based on the .NET handler's property settings.
+                // Since this is the first operation, we set all the necessary WinRT filter properties.
                 SetFilterProxyCredential();
                 SetFilterServerCredential();
+                if (_serverCertificateCustomValidationCallback != null)
+                {
+                    // The WinRT layer uses a different model for the certificate callback. The callback is
+                    // considered "extra" validation. We need to explicitly ignore errors so that the callback
+                    // will get called.
+                    //
+                    // In addition, the WinRT layer restricts some errors so that they cannot be ignored, such 
+                    // as "Revoked". This will result in behavior differences between UWP and other platforms.
+                    // The following errors cannot be ignored right now in the WinRT layer:
+                    //
+                    //     ChainValidationResult.BasicConstraintsError
+                    //     ChainValidationResult.InvalidCertificateAuthorityPolicy
+                    //     ChainValidationResult.InvalidSignature
+                    //     ChainValidationResult.OtherErrors
+                    //     ChainValidationResult.Revoked
+                    //     ChainValidationResult.UnknownCriticalExtension
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.Expired);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.IncompleteChain);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.InvalidName);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.RevocationFailure);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.RevocationInformationMissing);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.Untrusted);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.WrongUsage);
+                    _rtFilter.ServerCustomValidationRequested += RTServerCertificateCallback;
```

Comment:
>You can't let a connection transmit data if a custom callback didn't get to have its say.

This is a good point.  I think this demonstrates one example where we have to throw an exception at runtime if the light-up detection fails *and* we are trying to use that feature.

We can either throw an exception during the property setter of the callback. That would be a PNSE (PlatformNotSupportedException). Otherwise, if we validate everything during .SendAsync, we'll need to throw an HttpRequestException with perhaps a PNSE inside that.

I'm leaning toward throwing at the property setter since it is can be calculated right away that the feature won't work since we're running on a Windows 10 OS version before that callback was implemented in WinRT.

cc: @Diego-Perez-Botero @DavidGoll @CIPop